Sources/ComposableArchitecture/Internal/Create.swift

```diff
@@ -21,27 +21,23 @@
 // THE SOFTWARE.
 
 @preconcurrency import Combine
-import Darwin
+import Foundation
 
 final class DemandBuffer<S: Subscriber>: @unchecked Sendable {
   private var buffer = [S.Input]()
   private let subscriber: S
   private var completion: Subscribers.Completion<S.Failure>?
   private var demandState = Demand()
-  private let lock: os_unfair_lock_t
+  private let lock = NSRecursiveLock()
```

Comment:
Once again, I wanted to err on the side of caution and changed this to a recursive lock to avoid deadlocks — but it might be worth reasoning about the code more thoroughly and determining if it's possible to change this back. The main concern is that there are calls to `Subscriber.receive` while the lock is held, and afaik these can invoke arbitrary code that might call back into the `DemandBuffer` and deadlock us.